File: Selenium/Activities/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/target-practice")
    print("Title of the page is: ", driver.title)
    Third_header_text = driver.find_element(By.XPATH, "//h3[@id='third-header']").text
    print("Text of Third header on the page: ", Third_header_text)
    Fifth_header_element = driver.find_element(By.XPATH, "//h5[@class='ui green header']")
    color = Fifth_header_element.value_of_css_property("color")
    print("Color of fifth header on the page: ", color)
    violet_button = driver.find_element(By.CSS_SELECTOR, ".ui.violet.button")
    att_value = violet_button.get_attribute("class")
    print("Violet button classes: ", att_value)
    grey_button = driver.find_element(By.CSS_SELECTOR, ".ui.grey.button").text
    print("Grey button text: ", grey_button)

File: Selenium/Activities/Activity4.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity4 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/target-practice");
        Thread.sleep(2000);
        System.out.println("Title of the page: " + driver.getTitle());
        // Finding the 3rd header on the page and print it's text
        String thirdHeaderText = driver.findElement(By.xpath("//h3[@id='third-header']")).getText();
        System.out.println(thirdHeaderText);
        // Finding the 5th header on the page and print it's color
        String fifthHeaderElement = driver.findElement(By.xpath("//h5[@class='ui green header']")).getCssValue("color");
        System.out.println(fifthHeaderElement);
        // Finding the violet button and print all it's classes
        String violetButton = driver.findElement(By.cssSelector(".ui.violet.button")).getAttribute("class");
        System.out.println(violetButton);
        // Finding the grey button and print it's text
        String greyButton = driver.findElement(By.cssSelector(".ui.grey.button")).getText();
        System.out.println(greyButton);
        driver.quit();
    }
}
